plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -67,34 +67,40 @@ func (c *CachedNodeInfo) GetNodeInfo(id string) (*api.Node, error) {
 	return node.(*api.Node), nil
 }
 
+/**
+*  Note that predicateMetdata and matchingPodAntiAffinityTerm need to be declared in the same file
```

Comment:
not sure but, iirc it failed compilation when it was  in metadata.go (which is in the same predicates package) because theres some kind of parsing step that we do... is this a bug ?  i can double verify if necessary.